Contributing.md

```diff
@@ -61,3 +57,47 @@ and then push it.
 The CI contains credentials and will push artifacts to sonatype/gradlePortal when a tag is pushed. Every tag will trigger a new release.
 
 After a successful release, do not forget to add a changelog to the [releases page](https://github.com/apollographql/apollo-android/releases).
+
+
+## Overview of the CI
+
+The project uses [GitHub Actions](https://docs.github.com/en/actions) to automate the build process.
+
+We have [3 workflows](https://github.com/apollographql/apollo-android/tree/dev-3.x/.github/workflows), triggered by the following events:
+
+### On PRs
+
+**Workflow:** [`pr.yml`](https://github.com/apollographql/apollo-android/blob/dev-3.x/.github/workflows/pr.yml)
+
+**Jobs:**
+- `tests-gradle`
+    - Slow gradle tests
+- `tests-no-gradle`
+    - All root tests but not the Gradle ones and not the "exotic" apple ones (tvos, watchos)
+    - All apiCheck
+- `tests-integration`
+    - All integration tests
+
+### On pushes to `dev-3.x` branch
+
+_(Will be replaced to "on pushes to `main` branch" after v3 is released.)_
```

Comment:
I'm betting a pumkin pie that we're going to forget this and have docs out of sync with the code in a few weeks. 🙃

Reply:
You are probably right 😛 Created [this issue](https://github.com/apollographql/apollo-android/issues/3525) so we don't forget.